crates/router/src/core/payment_methods/cards.rs

```diff
@@ -2861,35 +2915,99 @@ pub async fn filter_payment_methods(
+                                    },
+                                )));
+                            };
+                        });
+
+                    payment_attempt
                         .and_then(|attempt| attempt.mandate_data.as_ref())
                         .map(|mandate_detail| {
                             if mandate_detail.update_mandate_id.is_some() {
-                                filter_pm_based_on_update_mandate_support_for_connector(
-                                    supported_payment_methods_for_update_mandate,
-                                    &payment_method,
-                                    &payment_method_object.payment_method_type,
-                                    connector_variant,
-                                )
-                            } else {
-                                true
+                                context_values.push(dir::DirValue::PaymentType(
+                                    euclid::enums::PaymentType::UpdateMandate,
+                                ));
+                                if let Ok(connector) = api_enums::RoutableConnectors::from_str(
+                                    connector_variant.to_string().as_str(),
+                                ) {
+                                    context_values.push(dir::DirValue::Connector(Box::new(
+                                        api_models::routing::ast::ConnectorChoice {
+                                            connector,
+                                            #[cfg(not(feature = "connector_choice_mca_id"))]
+                                            sub_label: None,
+                                        },
+                                    )));
+                                };
                             }
+                        });
+
+                    payment_attempt
+                        .map(|attempt| {
+                            attempt.mandate_data.is_none() && attempt.mandate_details.is_none()
                         })
-                        .unwrap_or(true);
+                        .and_then(|res| {
+                            res.then(|| {
+                                context_values.push(dir::DirValue::PaymentType(
+                                    euclid::enums::PaymentType::NonMandate,
+                                ))
+                            })
+                        });
 
-                    let filter9 = req
+                    payment_attempt
+                        .and_then(|inner| inner.capture_method)
+                        .and_then(|capture_method| {
+                            (capture_method == common_enums::CaptureMethod::Manual).then(|| {
+                                context_values.push(dir::DirValue::CaptureMethod(
+                                    common_enums::CaptureMethod::Manual,
+                                ));
+                            })
+                        });
```

Comment:
We are pushing `CaptureMethod` only if it is `Manual`, because we have `not_applicable_flows = manual` as the only variant in config. There is no place where we can find `not_applicable_flows = automatic`.

Moreover will using `then` in above code will produce some latency or something. Just curious to know why it should be converted to an `if`.

Reply:
Moreover it was done in the same way previously :)
<img width="518" alt="Screenshot 2024-05-23 at 17 52 34" src="https://github.com/juspay/hyperswitch/assets/61520228/f57cea91-5230-4006-aa07-b8c77eaabf90">

Reply:
> Moreover will using then in above code will produce some latency or something. Just curious to know why it should be converted to an if.

Using functional style code like this is more appropriate when your code is supposed to return a value (that's what the function signatures of `and_then()` and `then()` denote. Here we're not returning a value, we're performing an action with side effects, which is non-functional i.e. imperative, hence an `if` would be more suitable.